src/core/socket.h

```diff
@@ -164,6 +168,21 @@ struct Socket {
         RateLimit trigger_limit;
 };
 
+struct SocketPeer {
+        unsigned n_ref;
+
+        Socket *socket;
+
+        int family;
+        union sockaddr_union peer;
```

Comment:
given that we stick to sockaddr_union now, the separate "family" field is redundant, no? It's available as peer.sa.sa_family after all, too